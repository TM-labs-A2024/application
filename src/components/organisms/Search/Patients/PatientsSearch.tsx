import {
  Link,
  Button,
  Text,
  Heading,
  FormControl,
  FormLabel,
  Input,
  Stack,
  FormErrorMessage,
  HStack,
  Tag,
  TagLabel,
  TagCloseButton
} from '@chakra-ui/react'
import SearchInputComponent from '@components/atoms/SearchInput'
import PatientList from '@components/molecules/PatientsList'
import { FILTERS_APPLIED } from '@constants/index'
import { sexOptions, statusOptions, specialties } from '@constants/index'
import { ReactSelectOption, Patient } from '@src/types'
import { isIOS, isMobile } from '@utils/index'
import { formatDistanceToNowStrict } from 'date-fns'
import { es } from 'date-fns/locale/es'
import NextLink from 'next/link'
import React, { useEffect, useState, useCallback, useMemo } from 'react'
import { Controller, useForm, FieldErrors } from 'react-hook-form'
import { Store } from 'react-notifications-component'
import Select from 'react-select'

import Swimmer from '../../../../../public/static/icons/Swimmer.svg'

const sexOptionsList = sexOptions.map(({ id, name }: { id: string; name: string }) => ({
  value: id,
  label: name
}))

const statusOptionsList = statusOptions.map(({ id, name }: { id: string; name: string }) => ({
  value: id,
  label: name
}))

const specialtiesList = specialties.map(({ id, name }: { id: string; name: string }) => ({
  value: id,
  label: name
}))

type FormData = {
  sex: ReactSelectOption | null
  status: ReactSelectOption | null
  specialty: ReactSelectOption | null
  fromAge: string
  toAge: string
}

export default function PatientsSearch({
  context
}: {
  context: {
    approvedPatients: Patient[]
  }
}) {
  // --- Hooks -----------------------------------------------------------------
  const {
    handleSubmit,
    formState: { errors },
    control,
    register,
    watch,
    setValue
  } = useForm<FormData>()

  const values = watch()
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  const { approvedPatients } = context
  // --- END: Local state ------------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  const patientsFormatted = useMemo(
    () =>
      approvedPatients.map(
        ({ birthdate, govId, status, bed, firstname, lastname, pending, sex, specialties }) => ({
          href: pending ? '' : `/especialidades/${govId}`,
          title: `${firstname} ${lastname}`,
          description: `C.I: ${govId}, ${formatDistanceToNowStrict(new Date(birthdate), {
            locale: es,
            roundingMethod: 'floor'
          })}${status ? `, Cama: ${bed}` : ''}`,
          status,
          pending,
          sex,
          age: formatDistanceToNowStrict(new Date(birthdate), {
            locale: es,
            roundingMethod: 'floor'
          }).split(' ')[0],
          bed,
          specialties
        })
      ),
    [approvedPatients]
  )
  // --- END: Data and handlers ------------------------------------------------

  // --- Local state -----------------------------------------------------------
  const [showFilters, setShowFilters] = useState(false)
  const [patientsList, setPatientsList] = useState(patientsFormatted)
  const [search, setSearch] = useState('')
  const [fromAge, setFromAge] = useState('')
  const [toAge, setToAge] = useState('')
  const [sex, setsex] = useState<ReactSelectOption>({
    value: 0,
    label: ''
  })
  const [status, setStatus] = useState<ReactSelectOption>({
    value: 0,
    label: ''
  })
  const [specialty, setSpecialty] = useState<ReactSelectOption>({
    value: null,
    label: ''
  })
  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  const inputRef = React.createRef<HTMLInputElement>()
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------
  useEffect(() => {
    if (inputRef.current) {
      inputRef.current.focus()
    }
  }, [inputRef])

  useEffect(() => {
    let filteredPatientsList = patientsFormatted

    // Title filtering
    if (search !== '') {
      filteredPatientsList = filteredPatientsList.filter(
        (patient) =>
          patient.title.toLocaleLowerCase().includes(search.toLocaleLowerCase()) ||
          patient.description.toLocaleLowerCase().includes(search.toLocaleLowerCase())
      )
    }

    // sex filtering
    if (sex?.value) {
      filteredPatientsList = filteredPatientsList.filter((patient) => patient.sex === sex?.value)
    }

    // Status filtering
    if (status?.label) {
      filteredPatientsList = filteredPatientsList.filter((patient) => patient?.bed)
    }

    // specialty filtering
    if (specialty?.value) {
      filteredPatientsList = filteredPatientsList.filter((patient) =>
        patient?.specialties?.includes(String(specialty.value))
      )
    }

    // Dates filtering
    if (fromAge !== '') {
      filteredPatientsList = filteredPatientsList.filter(
        (patient) => Number(patient.age) >= Number(fromAge)
      )
    }

    if (toAge !== '') {
      filteredPatientsList = filteredPatientsList.filter(
        (patient) => Number(patient.age) <= Number(toAge)
      )
    }

    setPatientsList(filteredPatientsList)
  }, [
    approvedPatients,
    sex?.value,
    search,
    status?.label,
    specialty?.value,
    fromAge,
    toAge,
    patientsFormatted
  ])
  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  const removeFilters = useCallback(() => {
    setFromAge('')
    setToAge('')
    setsex({
      value: 0,
      label: ''
    })
    setStatus({
      value: 0,
      label: ''
    })
    setSpecialty({
      value: null,
      label: ''
    })
    setValue('fromAge', '')
    setValue('toAge', '')
    setValue('sex', null)
    setValue('status', null)
    setValue('specialty', null)
    setShowFilters(false)
  }, [setFromAge, setToAge, setsex, setStatus, setSpecialty, setValue])

  const onSubmit = (data: FormData) => {
    setFromAge(data.fromAge)
    setToAge(data.toAge)
    setsex(data?.sex)
    setStatus(data?.status)
    setSpecialty(data?.specialty)
    setShowFilters(false)
    Store.addNotification(FILTERS_APPLIED(isMobile(window)))
  }

  const verifyErrors = (errors: FieldErrors<FormData>) => Object.keys(errors).length > 0

  const matches = useMemo(() => `${patientsList.length} resultados`, [patientsList])

  const onChange = useCallback((value: string) => {
    setSearch(value)
  }, [])
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div
      className={`relative mx-auto block h-screen w-screen overflow-hidden px-8 lg:w-1/2 ${isIOS() ? 'pt-20' : 'pt-8'}`}
      data-testid="patients-search"
    >
      {showFilters && (
        <>
          <div className="mb-8 flex flex-row items-center justify-between gap-4">
            <Text>Ingresar filtros</Text>
            <Button variant="link" colorScheme="black" fontWeight="normal" onClick={removeFilters}>
              Cancelar
            </Button>
          </div>
          <Heading as="h2" size="md">
            Edad del paciente
          </Heading>
          <form onSubmit={handleSubmit(onSubmit)} className="h-full w-full">
            <FormControl isInvalid={verifyErrors(errors)} className="relative h-4/5 w-full">
              <div className="my-6 flex w-full flex-row">
                <div className="mr-2 flex w-1/2 flex-col">
                  <FormLabel>Desde</FormLabel>
                  <Input
                    id="fromAge"
                    className="min-h-10"
                    type="number"
                    placeholder="Desde"
                    defaultValue={fromAge}
                    {...register('fromAge', {
                      validate: {
                        ageFrom: (value) =>
                          values.toAge === null ||
                          values.toAge === '' ||
                          value <= values.toAge ||
                          'La edad de comienzo debe ser menor a la edad final'
                      }
                    })}
                  />
                  <FormErrorMessage>{errors?.fromAge && errors?.fromAge?.message}</FormErrorMessage>
                </div>
                <div className="ml-2 flex w-1/2 flex-col">
                  <FormLabel>Hasta</FormLabel>
                  <Input
                    id="toAge"
                    className="min-h-10"
                    type="number"
                    placeholder="Hasta"
                    defaultValue={toAge}
                    {...register('toAge', {
                      validate: {
                        ageTo: (value) =>
                          values.fromAge === null ||
                          values.fromAge !== '' ||
                          value >= values.fromAge ||
                          'La edad final debe ser mayor a la edad de comienzo'
                      }
                    })}
                  />
                  <FormErrorMessage>{errors?.toAge && errors?.toAge?.message}</FormErrorMessage>
                </div>
              </div>
              <Controller
                control={control}
                name="sex"
                render={({ field }) => (
                  <Select
                    id="sex"
                    {...field}
                    placeholder="Sexo"
                    options={sexOptionsList}
                    className="mb-6"
                  />
                )}
              />
              <Controller
                control={control}
                name="status"
                render={({ field }) => (
                  <Select
                    id="status"
                    {...field}
                    placeholder="Estado"
                    options={statusOptionsList}
                    className="mb-6"
                  />
                )}
              />
              <Controller
                control={control}
                name="specialty"
                render={({ field }) => (
                  <Select
                    id="specialty"
                    {...field}
                    placeholder="Especialidad"
                    options={specialtiesList}
                    className="mb-6"
                  />
                )}
              />
              <Stack mt={4} className="absolute bottom-0 w-full">
                <Button type="submit">Aplicar filtros</Button>
                <Button
                  variant="link"
                  colorScheme="black"
                  fontWeight="normal"
                  onClick={removeFilters}
                  className="mt-2"
                >
                  Quitar filtros
                </Button>
              </Stack>
            </FormControl>
          </form>
        </>
      )}
      {!showFilters && (
        <>
          <div className="mb-8 flex flex-row items-center">
            <SearchInputComponent
              placeholder="Buscar paciente"
              className="mr-4 w-full"
              onChange={onChange}
              inputRef={inputRef}
            />
            <Link as={NextLink} href="/pacientes">
              Cancelar
            </Link>
          </div>
          <div className="w-full overflow-x-scroll">
            <HStack spacing={4} mb={4} className="w-max">
              {fromAge && (
                <Tag size="md" variant="outline" colorScheme="blackAlpha">
                  <TagLabel>Desde: {fromAge}</TagLabel>
                  <TagCloseButton
                    onClick={() => {
                      setFromAge('')
                      setValue('fromAge', '')
                    }}
                  />
                </Tag>
              )}
              {toAge && (
                <Tag size="md" variant="outline" colorScheme="blackAlpha">
                  <TagLabel>Hasta: {toAge}</TagLabel>
                  <TagCloseButton
                    onClick={() => {
                      setToAge('')
                      setValue('toAge', '')
                    }}
                  />
                </Tag>
              )}
              {sex?.label && (
                <Tag size="md" variant="outline" colorScheme="blackAlpha">
                  <TagLabel>Sexo: {sex.label}</TagLabel>
                  <TagCloseButton
                    onClick={() => {
                      setsex(null)
                      setValue('sex', null)
                    }}
                  />
                </Tag>
              )}
              {status?.label && (
                <Tag size="md" variant="outline" colorScheme="blackAlpha">
                  <TagLabel>Estado: {status.label}</TagLabel>
                  <TagCloseButton
                    onClick={() => {
                      setStatus(null)
                      setValue('status', null)
                    }}
                  />
                </Tag>
              )}
              {specialty?.label && (
                <Tag size="md" variant="outline" colorScheme="blackAlpha">
                  <TagLabel>Especialidad: {specialty.label}</TagLabel>
                  <TagCloseButton
                    onClick={() => {
                      setSpecialty(null)
                      setValue('specialty', null)
                    }}
                  />
                </Tag>
              )}
            </HStack>
          </div>
          <div className="h-3/4 overflow-scroll">
            <PatientList pendingPatients={[]} patients={patientsList} label={matches} />
          </div>
          <div className="absolute bottom-16 w-full pr-16">
            <Button className="w-full gap-2" onClick={() => setShowFilters(!showFilters)}>
              <Swimmer /> Filtros
            </Button>
          </div>
        </>
      )}
    </div>
  )
}
