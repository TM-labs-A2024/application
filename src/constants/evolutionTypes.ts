export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
