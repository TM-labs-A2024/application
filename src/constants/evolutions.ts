export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
