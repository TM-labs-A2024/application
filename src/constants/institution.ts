export const institution = {
  id: '3fa85f64-5717-4562-b3fc-2c963f66afa1',
  name: 'Clínica Tex Mex',
  credentials: '14578245',
  address: 'Av. Fernandez Peña, 147.',
  email: 'test@test.com',
  phone: '+582742214441'
}
