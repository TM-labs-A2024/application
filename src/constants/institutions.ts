export const institutions = [
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa1',
    name: 'Clínica Tex Mex 1',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa11',
    name: 'Clínica Tex Mex 2',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa12',
    name: 'Clínica Tex Mex 3',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa13',
    name: 'Clínica Tex Mex 4',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa2',
    name: 'Clínica Tex Mex 5',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa3',
    name: 'Clínica Tex Mex 6',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa4',
    name: 'Clínica Tex Mex 7',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa5',
    name: 'Clínica Tex Mex 8',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa6',
    name: 'Clínica Tex Mex 9',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa7',
    name: 'Clínica Tex Mex 10',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa8',
    name: 'Clínica Tex Mex 11',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441',
    pending: true
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66afa9',
    name: 'Clínica Tex Mex 12',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af10',
    name: 'Clínica Tex Mex 13',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af11',
    name: 'Clínica Tex Mex 14',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af12',
    name: 'Clínica Tex Mex 15',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af13',
    name: 'Clínica Tex Mex 16',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af14',
    name: 'Clínica Tex Mex 17',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af15',
    name: 'Clínica Tex Mex 18',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af16',
    name: 'Clínica Tex Mex 19',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  },
  {
    id: '3fa85f64-5717-4562-b3fc-2c963f66af17',
    name: 'Clínica Tex Mex 20',
    credentials: '14578245',
    address: 'Av. Fernandez Peña, 147.',
    email: 'test@test.com',
    phone: '+582742214441'
  }
]
