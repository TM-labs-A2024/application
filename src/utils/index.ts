import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
