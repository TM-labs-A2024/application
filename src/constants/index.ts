import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
