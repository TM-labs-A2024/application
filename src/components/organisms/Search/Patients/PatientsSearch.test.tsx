import { patients } from '@src/constants'
import { render, screen, fireEvent, waitFor } from '@test/utils/index'

import '@testing-library/jest-dom'
import 'react-notifications-component'
import PatientsSearch from './PatientsSearch'

jest.mock('react-notifications-component')

const context = {
  approvedPatients: patients.filter((patient) => !patient.pending)
}

describe('Organisms > Search > PatientsSearch test', () => {
  test('The component shows the patients which match the entered criteria', () => {
    render(<PatientsSearch context={context} />)

    const input = screen.getByPlaceholderText('Buscar paciente')
    fireEvent.change(input, { target: { value: 'José Feliciano' } })

    expect(screen.getAllByText('José Feliciano Gutierrez García')).toHaveLength(7)
  })

  test('The component shows the filters if the filters button is clicked', () => {
    render(<PatientsSearch context={context} />)

    const button = screen.getByText('Filtros')
    fireEvent.click(button)

    expect(screen.getByText('Ingresar filtros')).toBeInTheDocument()
    expect(screen.getByText('Edad del paciente')).toBeInTheDocument()
    expect(screen.getByText('Aplicar filtros')).toBeInTheDocument()
    expect(screen.getByText('Quitar filtros')).toBeInTheDocument()
  })

  test('The component shows the patients which match the entered filters and shows the filters on top of the results', async () => {
    render(<PatientsSearch context={context} />)

    const button = screen.getByText('Filtros')
    fireEvent.click(button)

    const inputFrom = screen.getByPlaceholderText('Desde')
    fireEvent.change(inputFrom, { target: { value: '30' } })

    const inputTo = screen.getByPlaceholderText('Hasta')
    fireEvent.change(inputTo, { target: { value: '99' } })

    const submitButton = screen.getByText('Aplicar filtros')
    fireEvent.click(submitButton)

    const ages = ['30 años', '31 años', '32 años']

    await waitFor(() => {
      ages.forEach((age: string) => {
        const ageMatcher = (content: string) => content.includes(age)
        const ageElement = screen.getByText(ageMatcher)

        expect(ageElement).toBeInTheDocument()
      })
    })

    expect(screen.getByText('Desde: 30')).toBeInTheDocument()
    expect(screen.getByText('Hasta: 99')).toBeInTheDocument()
  })

  test('Matches the snapshot', () => {
    render(<PatientsSearch context={context} />)

    expect(screen.getByTestId('patients-search')).toMatchSnapshot()
  })
})
