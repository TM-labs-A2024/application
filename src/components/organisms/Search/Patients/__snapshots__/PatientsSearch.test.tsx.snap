// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > Search > PatientsSearch test Matches the snapshot 1`] = `
<div
  class="relative mx-auto block h-screen w-screen overflow-hidden px-8 lg:w-1/2 pt-8"
  data-testid="patients-search"
>
  <div
    class="mb-8 flex flex-row items-center"
  >
    <div
      class="mr-4 w-full"
      data-testid="search-input-component"
    >
      <div
        class="chakra-input__group css-1y0e7gb"
        data-group="true"
      >
        <div
          class="chakra-input__left-element css-gv1iwz"
        >
          <svg
            class="chakra-icon css-13izzit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <input
          class="chakra-input min-h-10 css-qluo93"
          placeholder="Buscar paciente"
        />
      </div>
    </div>
    <a
      class="chakra-link css-spn4bz"
      href="/pacientes"
    >
      Cancelar
    </a>
  </div>
  <div
    class="w-full overflow-x-scroll"
  >
    <div
      class="chakra-stack w-max css-hrntqi"
    />
  </div>
  <div
    class="h-3/4 overflow-scroll"
  >
    <div
      class="chakra-stack h-full css-8g8ihq"
      data-testid="patients-list"
    >
      <h2
        class="chakra-heading mb-4 css-9f6g39"
      >
        11 resultados
      </h2>
      <div
        class="overflow-auto"
      >
        <ul
          class="css-155za0w"
          role="list"
        >
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885563"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Josefa Feliciana Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885563, 27 años, Cama: B-67
                  </p>
                </div>
                <span
                  class="css-1eveppl"
                >
                  <svg
                    class="chakra-icon chakra-icon css-vvjt0k"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
                      fill="currentColor"
                    />
                  </svg>
                  <span
                    class="css-1ny2kle"
                  >
                    hospitalizado
                  </span>
                </span>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885564"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Feliciano Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885564, 28 años, Cama: C-66
                  </p>
                </div>
                <span
                  class="css-1eveppl"
                >
                  <svg
                    class="chakra-icon chakra-icon css-vvjt0k"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
                      fill="currentColor"
                    />
                  </svg>
                  <span
                    class="css-1ny2kle"
                  >
                    hospitalizado
                  </span>
                </span>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885565"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Josefa Feliciana Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885565, 29 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885566"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Feliciano Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885566, 30 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885567"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Feliciano Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885567, 31 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885568"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Josefa Feliciana Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885568, 32 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885569"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Feliciano Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885569, 26 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885560"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    Josefa Feliciana Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885560, 26 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885511"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Feliciano Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885511, 26 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885512"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Feliciano Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885512, 26 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="patient-card"
              href="/especialidades/E99885513"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Feliciano Gutierrez García
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: E99885513, 26 años
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="absolute bottom-16 w-full pr-16"
  >
    <button
      class="chakra-button w-full gap-2 css-ez23ye"
      type="button"
    >
      <div />
       Filtros
    </button>
  </div>
</div>
`
